### ISSUE TITLE
Access to Current/Previous Frame Element

### ISSUE THREAD

**Issue by ryanfitzer** [amio/re-carousel#111]
I'm looking for some way to manipulate attributes for the current and previous frames.

For example, I'd like to toggle `aria-hidden` to be true/false depending on if the frame is showing or not.

I could do it in a widget using `findDOMNode()` and the `index` prop, but I'm hoping there is a better way. Passing these as `ref`s to widgets is one idea. What do you think?

**Comment by amio** [comment]
I think this attribute `aria-hidden` should be built-in :D

**Comment by ryanfitzer** [comment]
I agree that built-in would be nice, but there may be other use cases that require access.

**Comment by amio** [comment]
I prefer not to expose DOM element to the api surface in a React component, if there's some use case do need this, we may do it the DOM-way.

**Comment by ryanfitzer** [comment]
The use case I’m looking to support is focusing the current frame after animation. This enables a screen reader to start reading immediately instead of the user having to move backward to find the element.

This could also be done in a different way by updating the proper`aria-` attribute. Not the same, but helpful. 

I understand your reluctance, The problem I see is maintaining a DOM reference during re-renders. Using a DOM method that returns a live `NodeList` would work, but not be as desirable as being passed the current frame as a `ref`. 

The other issue is that since there are no hooks for `[before/after]FrameAnimation` There’s no way to know when to access the current frame. Widgets `render()` method are fired many times for each change of frame.

**Comment by amio** [comment]
`[before/after]FrameAnimation` hooks would be nice to have.

> The use case I’m looking to support is focusing the current frame after animation. This enables a screen reader to start reading immediately instead of the user having to move backward to find the element.

I'll keep this in mind to see if I can make it easier while rewriting :D

**Comment by ryanfitzer** [comment]
Thanks for the addition!
